src/db_impl.cc

```diff
@@ -496,6 +507,32 @@ Options TitanDBImpl::GetOptions(ColumnFamilyHandle* column_family) const {
   return options;
 }
 
+Status TitanDBImpl::SetOptions(
+    ColumnFamilyHandle* column_family,
+    const std::unordered_map<std::string, std::string>& new_options) {
+  auto opts = new_options;
+  auto p = opts.find("blob_run_mode");
+  if (p != opts.end()) {
+    TitanBlobRunMode mode = TitanBlobRunMode::kNormal;
+    auto pm = blob_run_mode_string_map.find(p->second);
+    if (pm == blob_run_mode_string_map.end()) {
+      return Status::InvalidArgument("No blob-run-mode defined for " +
+                                     p->second);
+    } else {
+      mode = pm->second;
+    }
+    MutexLock l(&mutex_);
+    auto& table_factory = titan_table_factory_[column_family->GetID()];
+    table_factory->SetBlobRunMode(mode);
+    opts.erase(p);
+  }
+  if (opts.size() > 0) {
+    return db_->SetOptions(column_family, opts);
```

Comment:
Let's call `db_->SetOptions()` before handling `blob_run_mode`. In case `db_->SetOptions()` fail we should make the call fail as a whole.